server/api/admin.go

```diff
@@ -59,13 +59,18 @@ func (h *adminHandler) ResetTS(w http.ResponseWriter, r *http.Request) {
 	if err := apiutil.ReadJSONRespondError(h.rd, w, r.Body, &input); err != nil {
 		return
 	}
-	ts, ok := input["tso"].(float64)
-	if !ok || ts < 0 {
+	tsValue, ok := input["tso"].(string)
```

Comment:
Why not use `uint64`as the type?

Reply:
There is no such thing in js spec. 😕

Reply:
Considering this: https://play.golang.org/p/8-COmwYna7T

I think it is supported in go.

Reply:
I think it also works by define `var input map[string]uint64`.

Reply:
@5kbpers cc

Reply:
I'm afraid of it does not work in other languages, eg, rust, as it's not specified in json spec.

Reply:
@overvenus The language does not matter much, maybe the encoding library does. However, there should be no precision loss when encoding a number because that JSON is based on text not binary.

Reply:
If you encode a large number into JSON, it is just a sequence of digitals. IMO there is no reason for an encoding library to lose precision.